ext/mbstring/mbstring.c

```diff
@@ -3667,18 +3667,15 @@ php_mb_numericentity_exec(INTERNAL_FUNCTION_PARAMETERS, int type)
 
 	/* conversion map */
 	convmap = NULL;
-	if (Z_TYPE_P(zconvmap) == IS_ARRAY) {
-		target_hash = Z_ARRVAL_P(zconvmap);
-		i = zend_hash_num_elements(target_hash);
-		if (i > 0) {
-			convmap = (int *)safe_emalloc(i, sizeof(int), 0);
-			mapelm = convmap;
-			mapsize = 0;
-			ZEND_HASH_FOREACH_VAL(target_hash, hash_entry) {
-				*mapelm++ = zval_get_long(hash_entry);
-				mapsize++;
-			} ZEND_HASH_FOREACH_END();
-		}
+	i = zend_hash_num_elements(target_hash);
+	if (i > 0) {
+		convmap = (int *)safe_emalloc(i, sizeof(int), 0);
+		mapelm = convmap;
+		mapsize = 0;
+		ZEND_HASH_FOREACH_VAL(target_hash, hash_entry) {
+			*mapelm++ = zval_get_long(hash_entry);
+			mapsize++;
+		} ZEND_HASH_FOREACH_END();
 	}
 	if (convmap == NULL) {
```

Comment:
We should probably add a check that `mapsize % 4 == 0` ... right now we're just silently dropping trailing elements. I think we should also drop the `convmap == NULL` check here or the special-casing for empty convmap in general. If there is nothing to convert, it can just return the original string after all... Or if we want to treat it as an error, it should at least warn.

Reply:
ACK. I went with no special casing for empty `$convmap`, but throwing a `ValueError` for incomplete `$convmap` (the error message might be improvable).